velox/vector/arrow/Bridge.cpp

```diff
@@ -721,11 +729,42 @@ void exportMaps(
   Selection childRows(child.size());
   exportOffsets(vec, rows, out, pool, holder, childRows);
   holder.resizeChildren(1);
-  exportToArrowImpl(child, childRows, *holder.allocateChild(0), pool);
+  exportToArrowImpl(child, childRows, *holder.allocateChild(0), pool, options);
   out.n_children = 1;
   out.children = holder.getChildrenArrays();
 }
 
+template <TypeKind kind>
+void flattenAndExport(
+    const BaseVector& vec,
+    const Selection& rows,
+    ArrowArray& out,
+    memory::MemoryPool* pool,
+    VeloxToArrowBridgeHolder& holder) {
+  using NativeType = typename velox::TypeTraits<kind>::NativeType;
+  SelectivityVector allRows(vec.size());
+  DecodedVector decoded(vec, allRows);
+  auto flatVector = BaseVector::create<FlatVector<NativeType>>(
```

Comment:
Why not just call `BaseVector::flattenVector`?

Reply:
That function does in-place flattening of the input. The input is a const reference so we would have to  modify the input type. 
From a data perspective the input is a dictionary vector that wraps around the actual partitioned data. Doing an in-place change would likely affect the input data (I have not tried to change the code to try and see what would happen).

Reply:
It only does the in-place change when the reference is unique.  Otherwise a copy is created same as here.  But yeah with the current function signature it would be hard so we can keep it as it is for now.